bezier_library/include/bezier_library/bezier_params_observer.hpp

```diff
@@ -0,0 +1,124 @@
+#ifndef BEZIER_PARAMS_OBSERVER_HPP
+#define BEZIER_PARAMS_OBSERVER_HPP
+
+#include <bezier_library/bezier_library.hpp>
+
+/**
+ * Bezier parameters observer class
+ * This class allows to reduces computations done by the bezier library.
+ * It achieves that by comparing parameters of the library each time a trajectory generation
+ * is required. If the trajectory haven't change between two computation, it informs the trajectory
+ * generation function so the concerned operation is skipped.
+ */
+
+class BezierParamsObserver
+{
+
+public:
+
+  BezierParamsObserver()
+  {
+    // When a BezierParamObserver object is instantiated, all the values are set to their maximum
+    // numeric limits. This ensure that for the call to a bezier trajectory computation, all the
+    // operations will be perform because the the parameters of the library must never be equals
+    // to the numerics limits. After a first computation is performed, the observer will keep in memory those
+    // first parameters and will start comparing them with their new counterpart when other computation are performed.
+    setCurrentParams(std::numeric_limits<double>::max(), std::numeric_limits<unsigned>::max(),
```

Comment:
> This ensure that for the call to a bezier trajectory computation, all the operations will be perform because the the parameters of the library must never be equals to the numerics limits

This make sure that all the computations are done the first time the observer is requested.